### DISCUSSION TITLE
File upload changes the name of my files

### DISCUSSION THREAD

**Discussion by elfeffe** [livewire/livewire#1927] [4. General]
I'm trying to upload files using a "dropzone" (not dropzone.js, just using alpinejs), and it works well.
But when I try to save them, Livewire changes the name of my files.

This is my code (my property is $uploadingFiles = [] )
I use dispatchBrowserEvent because I show a toast with the error, and @error doesn't work.
I also use ValidationException because I can't get it to work using $validator->fails()

 ```
public function updatedUploadingFiles()
    {
        $error = false;
        try {
            $this->validate(['uploadingFiles.*' => config('lwfilemanager.upload_size')]);
        } catch (ValidationException $e)
        {
            $error = true;
            $this->dispatchBrowserEvent('file-upload', ['text' => config('lwfilemanager.upload_size_error_message'), 'type' => 'error']);
        }

        if(config('lwfilemanager.upload_type'))
        {
            try {
                $this->validate(['uploadingFiles.*' => config('lwfilemanager.upload_size')]);
            } catch (ValidationException $e)
            {
                $error = true;
                $this->dispatchBrowserEvent('file-upload', ['text' => config('lwfilemanager.upload_type_error_message'), 'type' => 'error']);
            }
        }
        
        if(!$error)
        {
            foreach ($this->uploadingFiles as $file)
            {
                try
                {
                    $file->store($this->currentDirectory, $this->disk);

                } catch (\Exception $e)
                {
                    $this->dispatchBrowserEvent('file-upload', ['text' => $e->getMessage(), 'type' => 'error']);
                }
            }
        }
    }

```

But my file 2.png is converted to RQhyYHv3ccAhOEAj5NCztzzwARpi027o5BUpkuiZ.png

What am I doing wrong?

**Comment by joshhanley** [comment]
Under the hood Livewire is just using Laravel's file storage class, which by default converts the filename to a hash, so you don't get an collisions in your storage directory. 

You can either use `$file->storeAs...` to specify the file name yourself see https://laravel.com/docs/8.x/filesystem#specifying-a-file-name

Or you can store it with the has and get the original file name using `$file->getClientOriginalName();` see https://laravel.com/docs/8.x/filesystem#other-file-information

Hope this helps!

**Comment by elfeffe** [comment]
@joshhanley Hey! You saved my day!
Thank you!
